libs/api-bundle/src/Security/StorageApiToken/StorageApiToken.php

```diff
@@ -94,6 +94,6 @@ public function getRoles(): array
 
     public function getAllowedComponents(): ?array
     {
-        return $this->tokenInfo['componentAccess'] ?? [];
```

Comment:
Tohle bylo špatně!

Token, který u sebe nemá vůbec componentAccess = má oprávnění na všechny komponenty.
Pokud má u sebe prázdné pole tak nemá oprávnění na žádnou.

Reply:
Tim padem je bohuzle nazev takovy zavadejici, ale asi se nam ted nechce vymyslet neco lepsiho.

Mozna apson pridat metodu `canAccessComponent`? Sice tu neni nikde direct potreba, ale rekl bych, ze to bude hlavni use-case v mistech, kde se pouziva token a kdyz tu bude, navede te to k jejimu pouziti, nez aby sis vzpominal, ze `null` ma specialni vyznam? (predpokladam, ze tuhle botu jsme tu udelal ja prave proto, ze mi ten `null` nedosel)

```
public function canAccessComponent(string $componentId): bool
{
    return in_array($componentId, $this->getAllowedComponents() ?? [], true);
}
```

Reply:
No zavádějící to je, ale jak sis to připravil, takové to máš:)

Používá se to tu:
https://github.com/keboola/platform-libraries/blob/main/libs/permission-checker/src/StorageApiToken.php#L12-L23

A ta nová metoda mi nedává smysl, máš ji až v checkeru, kde se i používá. Dokonce jí tam máš napsanou i správně, tak jak má být
https://github.com/keboola/platform-libraries/blob/main/libs/permission-checker/src/StorageApiToken.php#L48-L51

Reply:
No tohle je neco jineho, nez token v permission checkeru - v permission checkeru je navrzeny tak, aby se delaly snando permission checky.. tohle je token, ktery dostanes jako `#[CurrentUser]` v controlleru. Ale ok, nechme to ted tedy byt.

Reply:
No ale tohle co navrhujes se pro nic nepouzije.

Reply:
Ano, to jsem popsal a zduvodnil :slightly_smiling_face: Ale rikam, nech to ted byt, to byl spis navrh na vylepseni, ne blocker.